mat/index_bound_checks.go

```diff
@@ -346,3 +346,52 @@ func (d *DiagDense) setDiag(i int, v float64) {
 	}
 	d.mat.Data[i*d.mat.Inc] = v
 }
+
+// At returns the element at row i, column j.
+func (a *Tridiag) At(i, j int) float64 {
+	return a.at(i, j)
+}
+
+func (a *Tridiag) at(i, j int) float64 {
+	if uint(i) >= uint(d.mat.N) {
+		panic(ErrRowAccess)
+	}
+	if uint(j) >= uint(d.mat.N) {
+		panic(ErrColAccess)
+	}
+	switch i - j {
+	case -1:
+		return a.mat.DU[i]
+	case 0:
+		return a.mat.D[i]
+	case 1:
+		return a.mat.DL[j]
+	default:
+		return 0
+	}
```

Comment:
This is elegant.

Reply:
Thanks. Btw, I didn't know it's possible to comment on a range of lines, nice.